drivers/adc/adc_stm32.c

```diff
@@ -1087,7 +1087,9 @@ static int start_read(const struct device *dev,
 	 */
 	adc_stm32_enable(adc);
 
+#if !defined(CONFIG_SOC_SERIES_STM32F1X)
```

Comment:
Note, there's still two OVR instances in `dma_callback` which needs to be handled. This build isn't failing because there's no test build with the STM32F1+ADC+DMA. It's done correctly in my PR, but I can also close it in favour if this one if this comment is addressed.

Reply:
Use the PR https://github.com/zephyrproject-rtos/zephyr/pull/56017